File: gradle/packages/cli-function/src/main/kotlin/com/app/cli/function/Details.kt
package com.app.cli.function

import com.app.hello.world.printDate
import com.app.translator.translate

object Details {
    @JvmStatic
    fun main(args: Array<String>) {
        println("===================================================")
        printUserDetails(args)
        println("===================================================")
    }

    private fun getUserName(args: Array<String>): String {
        return if (args.size <= 1) {
            print("Please enter your name : ")
            readLine()!!.trim()
        } else {
            args.sliceArray(1 until args.size).joinToString(" ")
        }
    }

    private fun getUserLanguage(): String {
        print("Please enter your language : ")
        return readLine()!!.trim()
    }

    fun printUserDetails(args: Array<String>) {
        val name = getUserName(args)
        val language = getUserLanguage()
        println(translate(language, name))
        println(printDate())
    }
}

File: dotnet/src/CLI.Function/Details/Details.cs
using Prog.Hello.World;
using Prog.Translation;

namespace Prog.CLI.Function;

public class Details
{
    public static string GetUserName(string[] args)
    {
        if (args == null || args.Length == 0)
        {
            Console.Write("Please enter your name : ");
            return Console.ReadLine()?.Trim() ?? string.Empty;
        }
        else
        {
            return string.Join(" ", args);
        }
    }

    public static string GetUserLanguage()
    {
        Console.Write("Please enter your language : ");
        return Console.ReadLine()?.Trim() ?? string.Empty;
    }

    public static void PrintUserDetails(string[] args)
    {
        string name = GetUserName(args);
        string language = GetUserLanguage();
        Console.WriteLine(Translator.Translate(language, name));
        Console.WriteLine(HelloWorld.PrintDate());
    }
}
